### PR TITLE
[INLONG-7688][Manager] Creating schema of StreamSource by SQL

### LINKED ISSUE #7688: [Feature][Manager] Creating schema of StreamSource by SQL
### Description

image

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
### Prepare a Pull Request
*[INLONG-7688][Manager] Creating schema of StreamSource by SQL*

- Fixes #7688 

image

### CODE REVIEW

**Comment by fuweng11** [comment]
@featzhang At the same time, please modify StreamSinkApi parseFields and InlongStreamApi.parseFields.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/StreamSinkController.java [hunk 1]
```diff
@@ -118,11 +118,14 @@ public Response<Boolean> deleteByKey(@RequestParam(required = false, defaultValu
...
-    @ApiImplicitParam(name = "fieldsJson", dataTypeClass = String.class, required = true)
-    public Response<List<SinkField>> parseFields(@RequestBody String fieldsJson) {
-        return Response.success(sinkService.parseFields(fieldsJson));
+    @RequestMapping(value = "/sink/parseFields/{statementType}", method = RequestMethod.POST)
+    @ApiOperation(value = "parse stream sink fields from statement")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "statement", dataTypeClass = String.class, required = true),
+            @ApiImplicitParam(name = "statementType", dataTypeClass = String.class, required = true)
+    })
+    public Response<List<SinkField>> parseFields(@PathVariable String statementType, @RequestBody String statement) {
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/StreamSinkController.java:127]: Can we encapsulate all the parameters into an object without displaying the parameters in the path?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java [hunk 2]
```diff
@@ -677,24 +692,64 @@ public Boolean updateAfterApprove(List<SinkApproveDTO> approveList, String opera
...
+            CreateTable createTable = (CreateTable) statement;
+            List<ColumnDefinition> columnDefinitions = createTable.getColumnDefinitions();
+            // get column definition
+            for (ColumnDefinition definition : columnDefinitions) {
+                // get field name
+                String columnName = definition.getColumnName();
+                ColDataType colDataType = definition.getColDataType();
+                String sqlDataType = colDataType.getDataType();
+                // Convert SQL type to TypeInfo
+                FormatInfo formatInfo = FieldInfoUtils.convertFieldFormat(sqlDataType, "");
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java:734]: Is this field conversion valid？As far as I know `FieldInfoUtils.convertFieldFormat` method is turning sourceFieldType into sinkFIeldType. However, the field type in the table creation statement you passed in belongs to the `sinkFieldType`.

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**